pennylane_lightning/lightning_qubit.py

```diff
@@ -74,6 +74,30 @@ class LightningQubit(DefaultQubit):
         "CRot",
     }
 
+    kernel_operations = {
+        "PauliX",
+        "PauliY",
+        "PauliZ",
+        "Hadamard",
+        "Hadamard",
+        "S",
+        "T",
+        "RX",
+        "RY",
+        "RZ",
+        "PhaseShift",
+        "Rot",
+        "CNOT",
+        "SWAP",
+        "CZ",
+        "CRX",
+        "CRY",
+        "CRZ",
+        "CRot",
+        "Toffoli",
+        "CSWAP",
+    }
```

Comment:
These are the gates that have a kernel supported. Ideally, we would not explicitly set `operations` and just let it inherit from `default.qubit`. We'd then have a check in `apply_lightning()` that maps the operations to `QubitUnitary` if the operation is not in `kernel_operations`. In such a way, we're supporting essentially all ops, even without a kernel.